### PR TITLE
Introducing swagger MSF4J support for swagger code generator

### PR DESCRIPTION
**Pull request by sanjeewa-malalgoda**
### PR checklist

- [x] Read the [contribution guildelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell/batch script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates)
- [x] Filed the PR against the correct branch: master for non-breaking changes and `2.3.0` branch for breaking (non-backward compatible) changes.

### Description of the PR

(details of the change, additional tests that have been done, reference to the issue for tracking, etc)

### CODE REVIEW

**Comment by sanjeewa-malalgoda** [comment]
This commit is to provide MSF4J(https://github.com/wso2/msf4j) support for swagger code generator. However ./bin/{LANG}-petstore.sh did not added as this is entirely new framework. Please let me know if anything need. I tried different swagger definitions including pet store and everything worked smoothly for me. Thank you all.

**Comment by wing328** [comment]
@sanjeewa-malalgoda thank you for your contribution. I'll review and let you know if I've any question.

**Comment by sanjeewa-malalgoda** [comment]
Also following text need to add documents section and it explains how we can generate code and run micro service.

Java Micro Services Framework For Java
MSF4J generator uses java-msf4j as the default library
java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate \
  -i http://petstore.swagger.io/v2/swagger.json \
  -l msf4j \
  -o samples/server/petstore/msf4j

Before you build/run service replace .deploy(new PetApi()) with your actual service class name in Application.java file like .deploy(new ApisAPI()) then it will start that service. If you have multiple service classes add them in , separated manner.

        new MicroservicesRunner()
                .deploy(new PetApi())
                .start();
    }

To Use-it : in the generated folder try mvn package for build jar.
Start your server java -jar target/micro-service-server-1.0.0.jar
Java Microservice listening on default port 8080

Run the following command or simply go to http://127.0.0.1:8080/pet/12 from your browser.
curl http://127.0.0.1:8080/pet/12

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaMSF4JServerCodegen.java [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+    protected static final String LIBRARY_JERSEY1 = "jersey1";
+    protected static final String LIBRARY_JERSEY2 = "jersey2";
+
+
+    public static final String DEFAULT_LIBRARY = LIBRARY_JERSEY2;
+
+    public JavaMSF4JServerCodegen() {
+        super();
+        outputFolder = "generated-code/JavaJaxRS-Jersey";
```
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaMSF4JServerCodegen.java]: Should we set `outputFolder` to "generated-code/JavaMS4J" instead?

Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaMSF4JServerCodegen.java]: Actually I found some other Java server stub generators do not have proper outputFolder set.

I'll file a PR to clean those up (including this one)

**Comment by wing328** [comment]
I've added the documentation to https://github.com/swagger-api/swagger-codegen/wiki/Server-stub-generator-HOWTO#java-msf4j

Please also update https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/main/resources/MSF4J/README.mustache as the instruction does not seem correct for MSF4J.

**Comment by wing328** [comment]
Tweet: https://twitter.com/wing328/status/791132415762784256

To all, please help retweet to promote the new generator.

**Comment by sanjeewa-malalgoda** [comment]
Thank you very much @wing328. It was great pleasure to work in this project and i hope to contribute further in future.

**Comment by wing328** [comment]
@sanjeewa-malalgoda Looking forward to your contributions in the future.

Btw, I've added you as the template owner of MSF4J in https://github.com/swagger-api/swagger-codegen/blob/master/README.md#template-creator

File: modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaMSF4JServerCodegen.java (hunk 1 above)
Comment (wing328) [inline modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaMSF4JServerCodegen.java]: FYI. https://github.com/swagger-api/swagger-codegen/pull/4076 submitted

**Pull request merged**